src/buildApplication/checkPackageManager.ts

```diff
@@ -0,0 +1,50 @@
+import YAML from 'js-yaml';
+import { spawn } from 'child_process';
+import { readFile } from 'fs/promises';
+import { validateFile } from '../utils';
+import { cliError } from '../cli';
+
+export async function checkPackageManager() {
+	const userAgent = process.env.npm_config_user_agent;
+
+	const hasYarnLock = await validateFile('yarn.lock');
+	const hasPnpmLock = await validateFile('pnpm-lock.yaml');
+
+	if ((userAgent && userAgent.startsWith('pnpm')) || hasPnpmLock) return 'pnpm';
+
+	if ((userAgent && userAgent.startsWith('yarn')) || hasYarnLock) {
+		const yarn = process.platform === 'win32' ? 'yarn.cmd' : 'yarn';
+		const getYarnV = spawn(yarn, ['-v']);
+		let yarnV = '';
+		getYarnV.stdout.on('data', data => {
+			yarnV = `${data}`.trimEnd();
+		});
+		getYarnV.stderr.on('data', data => {
+			cliError(data);
+		});
+		await new Promise((resolve, reject) => {
+			getYarnV.on('close', code => {
+				if (code === 0) {
+					resolve(null);
+				} else {
+					reject();
+				}
+			});
+		});
+		if (!yarnV.startsWith('1.')) {
+			await validateFile('.yarnrc.yml');
+			const yarnYAML = YAML.load(await readFile('.yarnrc.yml', 'utf-8')) as {
+				nodeLinker: 'node-modules' | string;
+			};
```

Comment:
nit, could we separate the lines?
```suggestion
                         const yarnrc = await readFile('.yarnrc.yml', 'utf-8');
			const yarnYAML = YAML.load(yarnrc) as {
				nodeLinker: 'node-modules' | string;
			};
```

also is there a benefit in `'node-modules' | string`?

```suggestion
                         const yarnrc = await readFile('.yarnrc.yml', 'utf-8');
			const yarnYAML = YAML.load(yarnrc) as {
				nodeLinker: string;
			};
```